src/System.Memory/tests/Performance/Perf.ReadOnlySequence.TryGet.cs

```diff
@@ -0,0 +1,374 @@
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void Char_MultiSegment(int size, int start, int lenght)
+        {
+            var segment1 = new BufferSegment<char>(new char[size]);
+            BufferSegment<char> segment2 = segment1;
+            for (int j = 0; j < 10; j++)
+                segment2 = segment2.Append(new char[size]);
+            var buffer = new ReadOnlySequence<char>(segment1, start, segment2, lenght);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        var p = buffer.Start;
+                        while (buffer.TryGet(ref p, out ReadOnlyMemory<char> memory))
+                            localInt ^= memory.Length;
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void String(int size, int start, int lenght)
+        {
+            ReadOnlyMemory<char> strMemory = new string('a', size).AsMemory();
+            strMemory = strMemory.Slice(start, lenght);
+            var buffer = new ReadOnlySequence<char>(strMemory);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < size / 10; j++)
+                        {
+                            var p = buffer.Start;
+                            while (buffer.TryGet(ref p, out ReadOnlyMemory<char> memory))
+                                localInt ^= memory.Length;
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Char_Empty()
+        {
+            ReadOnlySequence<char> buffer = ReadOnlySequence<char>.Empty;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
```

Comment:
Why two loops over Benchmark.InnerIterationCount?

Reply:
One loop was too fast. I added second loop.

Reply:
If a single loop is too fast, increase the value of InnerCount (`private const int InnerCount = 1000;`)